docs/writing-benchmarks.md

```diff
@@ -0,0 +1,236 @@
+# Writing Benchmarks
+
+If you're familiar with the Java Microbenchmark Harness (JMH) toolkit, you'll find that the `kotlinx-benchmark`
+library shares a similar approach to crafting benchmarks. This compatibility allows you to seamlessly run your
+JMH benchmarks written in Kotlin on various platforms with minimal, if any, modifications.
+
+Like JMH, kotlinx-benchmark is annotation-based, meaning you configure benchmark execution behavior using annotations.
+The library then extracts metadata provided through annotations to generate code that benchmarks the specified code
+in the desired manner.
+
+To get started, let's examine a simple example of a multiplatform benchmark:
+
+```kotlin
+import kotlinx.benchmark.*
+
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Warmup(iterations = 20, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 20, time = 1, timeUnit = TimeUnit.SECONDS)
+@BenchmarkTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+class ExampleBenchmark {
+
+    @Param("4", "10")
+    var size: Int = 0
+
+    private val list = ArrayList<Int>()
+
+    @Setup
+    fun prepare() {
+        for (i in 0..<size) {
+            list.add(i)
+        }
+    }
+
+    @TearDown
+    fun cleanup() {
+        list.clear()
+    }
+
+    @Benchmark
+    fun benchmarkMethod(): Int {
+        return list.sum()
+    }
+}
+```
+
+**Example Description**:
+This example tests the speed of summing numbers in an `ArrayList`. We evaluate this operation with lists
+of 4 and 10 numbers to understand the method's performance with different list sizes.
+
+## Explaining the Annotations
+
+The following annotations are available to define and fine-tune your benchmarks.
+
+### @State
+
+The `@State` annotation is used to mark benchmark classes.
```

Comment:
This paragraph gives a good overview of the `State`'s purpose but also leaves an impression that that purpose changes depending on a target platform.

I think we can write that it always `specifies the extent to which the state object is shared among the worker threads` and by default, it's mandatory. However, right now, the multi-threaded execution is only supported on JVM and there the scope has some other values. Also, when writing a JVM-only benchmark, that annotation could be omitted.

Reply:
Done. PTAL if it's better now.